pyiron_base/jobs/flex/pythonfunctioncontainer.py

```diff
@@ -64,21 +66,28 @@ def __call__(self, *args, **kwargs):
     def to_hdf(self, hdf=None, group_name=None):
         super().to_hdf(hdf=hdf, group_name=group_name)
         self.project_hdf5["function"] = np.void(cloudpickle.dumps(self._function))
+        self.project_hdf5["_mangle_name_on_save"] = self._mangle_name_on_save
 
     def from_hdf(self, hdf=None, group_name=None):
         super().from_hdf(hdf=hdf, group_name=group_name)
         self._function = cloudpickle.loads(self.project_hdf5["function"])
+        self._mangle_name_on_save = bool(self.project_hdf5["_mangle_name_on_save"])
 
     def save(self):
-        job_name = self._function.__name__ + get_hash(
-            binary=cloudpickle.dumps(
-                {"fn": self._function, "kwargs": self.input.to_builtin()}
+        if self._mangle_name_on_save:
+            job_name = self._function.__name__ + get_hash(
+                binary=cloudpickle.dumps(
+                    {"fn": self._function, "kwargs": self.input.to_builtin()}
+                )
             )
-        )
-        self.job_name = job_name
-        if job_name in self.project.list_nodes():
-            self.from_hdf()
-            self.status.finished = True
+
+            self.job_name = job_name
+
+            if self.job_name in self.project.list_nodes():
+                self.from_hdf()
+                self.status.finished = True
+            else:
+                super().save()
         else:
             super().save()
```

Comment:
```suggestion
        super().save()
```

Reply:
This breaks the behaviour as we no longer avoid saving when the job is already found, but I agree it's more readable so I've committed it and will add an explicit `return` to recover the correct behaviour.